src/qz/utils/JsonWriter.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * @author Brett B.
+ *
+ * Copyright (C) 2019 QZ Industries, LLC
+ *
+ * LGPL 2.1 This is free software.  This software and source code are released under
+ * the "LGPL 2.1 License".  A copy of this license should be distributed with
+ * this software. http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+
+
+package qz.utils;
+
+import org.apache.commons.io.Charsets;
+import org.apache.commons.io.FileUtils;
+import org.codehaus.jettison.json.JSONArray;
+import org.codehaus.jettison.json.JSONException;
+import org.codehaus.jettison.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Iterator;
+
+/**
+ * A minimally intrusive JSON writer
+ */
+public class JsonWriter {
+    protected static final Logger log = LoggerFactory.getLogger(JsonWriter.class);
+
+    public static boolean write(String path, String data, boolean overwrite, boolean delete) throws IOException, JSONException {
+        File f = new File(path);
+        if(!f.getParentFile().exists()) {
+            log.warn("Warning, the parent folder of {} does not exist, skipping.", path);
+            return false;
+        }
+
+        if (data == null) {
+            log.warn("Data is null, nothing to merge");
+            return true;
+        }
+
+        JSONObject config = f.exists() ? new JSONObject(FileUtils.readFileToString(f, Charsets.UTF_8)) : new JSONObject();
+        JSONObject append = new JSONObject(data);
+
+        if (!delete) {
+            merge(config, append, overwrite);
+        } else {
+            remove(config, append);
+        }
+
+        FileUtils.write(f, config.toString(2));
+
+        return true;
+    }
+
+    /**
+     * Appends all keys from {@code merger} to {@code base}
+     *
+     * @param base      Root JSON object to merge into
+     * @param merger    JSON Object of keys to merge
+     * @param overwrite If existing keys in {@code base} should be overwritten if defined in {@code merger}
+     */
+    private static void merge(JSONObject base, JSONObject merger, boolean overwrite) throws JSONException {
+        Iterator itr = merger.keys();
+        while(itr.hasNext()) {
+            String key = (String)itr.next();
+
+            Object baseVal = base.opt(key);
+            Object mergeVal = merger.opt(key);
+
+            if (baseVal == null) {
+                //add new key
+                base.put(key, mergeVal);
+            } else if (baseVal instanceof JSONObject && mergeVal instanceof JSONObject) {
+                //deep copy sub-keys
+                merge((JSONObject)baseVal, (JSONObject)mergeVal, overwrite);
+            } else if (overwrite) {
+                //force new key val if existing and allowed
+                base.put(key, mergeVal);
+            } else if (baseVal instanceof JSONArray && mergeVal instanceof JSONArray) {
```

Comment:
I believe you mentioned a repeat entry bug in arrays for this I still need to fix. 
Adding a comment so I remember to look into it.

Reply:
Pushed a fix
In regards to the formatting anomaly you mentioned before, it looks like a bug in the library when only 1 key is present in an object, they have a special n==1 case that forgets to add newlines.. Everything formats correctly for any objects with 2 or more keys.

Reply:
Did you test your theory?  I added another element, and this is what I got... It's still not formatting properly.

```json
{
  "policies": {"Certificates": {"ImportEnterpriseRoots": true}},
  "indentFix": -99999
}
```

Then I wrapped it in an array and it's still wrong... 

```json
[
  {"policies": {"Certificates": {"ImportEnterpriseRoots": true}}},
  {"indentFix": -99999}
]
```

Reply:
Nevermind, editing the class proves the original point.  Working on a patch now...

Reply:
Updated Jettison with a custom patch.  Indentation looks good now.  I just removed the `length() == 1` block completely. 

Sent request to both upstream projects (neither appear to be maintained)
https://github.com/codehaus/jettison/pull/10
https://github.com/jettison-json/jettison/pull/31